### PR TITLE
PR: Increase type safety for angle-related API

### PR DESCRIPTION
**Pull request by jkalias**
The implementation for issue #35

### CODE REVIEW

File: Sources/Angle.swift [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+
+    static var piHalf = Angle(degrees: 90)
+
+    static var pi = Angle(degrees: 180)
+
+    static var twoPi = Angle(degrees: 360)
+}
+
+public extension Angle {
+    var cos: Double { os.cos(radians) }
```
Comment (nicklockwood) [inline Sources/Angle.swift:47]: Semantically I don't think these should be properties since they are applying a transformation.

**Review by nicklockwood — CHANGES_REQUESTED** [review]
looks good. A few minor stylistic concerns but nothing major.

File: Sources/Angle.swift [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+
+#if os(Linux)
+import Glibc
+#else
+import Darwin
+#endif
+
+/// A type-safe struct for all API related to angles
+public struct Angle {
+    public let degrees: Double
```
[the remark this answers is not in the document]
Comment (jkalias) [reply Sources/Angle.swift:19]: This is a legit question. I thought about it as well, I guess the reason was that degrees are more human-friendly to read.

But you are right, I'll change the default initializer to radians.

File: Sources/Angle.swift [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+        return lhs.degrees < rhs.degrees
+    }
+}
+
+extension Angle: Hashable {}
+
+private struct os {
+    static func cos(_ radians: Double) -> Double {
+        #if os(Linux)
+        return Glibc.cos(radians)
```
[the remark this answers is not in the document]
Comment (jkalias) [reply Sources/Angle.swift:98]: Ok, I will change the ```Darwin.cos``` calls etc. to ```Foundation.cos```, but how would you like the Linux variant?

File: Sources/Angle.swift (hunk 1 above)
Comment (jkalias) [reply Sources/Angle.swift:47]: My intention was explicit, so they don't read like function calls but as properties. Similar to mathematics that a given angle _has_ a cosine etc.

However, you have the final word on this, so let me know how you'd like it.

**Comment by jkalias** [comment]
@nicklockwood since this is my first open source PR, is there something _I_ need to do now, or are you still thinking about it?

**Comment by nicklockwood** [comment]
@jkalias lgtm. Thanks!

**Pull request merged**